### PR TITLE
fix: ibc quota check enabled

### PR DESCRIPTION
**Pull request by gsk967**
## Description

closes: #XXXX

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] added appropriate labels to the PR
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/umee-network/umee/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items._

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

**Review by robert-zaremba — COMMENTED** [review]
1. If we are doing it then let's handle all cases . For example in `SendPacket`
2. We should also add tests.

File: x/uibc/quota/ibc_module.go [hunk 1]
```diff
@@ -41,17 +41,23 @@ func (im IBCMiddleware) OnAcknowledgementPacket(ctx sdk.Context, packet channelt
...
 }
 
 // OnTimeoutPacket implements types.Middleware
 func (im IBCMiddleware) OnTimeoutPacket(ctx sdk.Context, packet channeltypes.Packet, relayer sdk.AccAddress) error {
-	err := im.RevertQuotaUpdate(ctx, packet.Data)
-	emitOnRevertQuota(&ctx, "timeout", packet.Data, err)
+	params := im.keeper.GetParams(ctx)
+	if params.IbcStatus == uibc.IBCTransferStatus_IBC_TRANSFER_STATUS_QUOTA_ENABLED {
+		err := im.RevertQuotaUpdate(ctx, packet.Data)
+		emitOnRevertQuota(&ctx, "timeout", packet.Data, err)
```
Comment (robert-zaremba) [inline x/uibc/quota/ibc_module.go:59]: if there is a timeout then we should correctly revert the quota update, without checking the status.

**Review by robert-zaremba — COMMENTED** [review]
+ add a task to add tests.

File: x/uibc/quota/ibc_module.go (hunk 1 above)
Comment (gsk967) [reply x/uibc/quota/ibc_module.go:59]: Why don't check the status ?

**Comment by gsk967** [comment]
> * add a task to add tests.

I added task for this https://linear.app/umee/issue/UMEE-486/add-tests-to-ibc-status-param-enabled-to-check-quota

File: x/uibc/quota/ibc_module.go (hunk 1 above)
Comment (robert-zaremba) [reply x/uibc/quota/ibc_module.go:59]: Because we are in the timeout.

File: x/uibc/quota/keeper/relay.go [hunk 2]
```diff
@@ -24,8 +25,11 @@ func (k Keeper) SendPacket(ctx sdk.Context,
 		return 0, errors.Wrap(err, "bad packet in rate limit's SendPacket")
 	}
 
-	if err := k.CheckAndUpdateQuota(ctx, denom, funds); err != nil {
-		return 0, errors.Wrap(err, "bad packet in rate limit's SendPacket")
+	params := k.GetParams(ctx)
+	if params.IbcStatus == uibc.IBCTransferStatus_IBC_TRANSFER_STATUS_QUOTA_ENABLED {
+		if err := k.CheckAndUpdateQuota(ctx, denom, funds); err != nil {
+			return 0, errors.Wrap(err, "bad packet in rate limit's SendPacket")
```
Comment (robert-zaremba) [inline x/uibc/quota/keeper/relay.go:31]:
```suggestion
			return 0, errors.Wrap(err, "SendPacket over the IBC Quota")
```

**Review by robert-zaremba — APPROVED** [review]

**Pull request merged**